packages/modal/components/ModalBase.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from "react";
+import { cx } from "emotion";
+import FocusTrap from "focus-trap-react";
+import { FocusTarget } from "focus-trap";
+import { Transition } from "react-transition-group";
+import {
+  modal,
+  modalWidth,
+  modalPreTransitionStyle,
+  modalTransitionStyles,
+  scrim,
+  scrimPreTransitionStyle,
+  scrimTransitionStyles
+} from "../style";
+import ModalContents from "./ModalContents";
+import Overlay from "../../shared/components/Overlay";
+import { flex } from "../../shared/styles/styleUtils";
+
+export enum ModalSizes {
+  S = "s",
+  M = "m",
+  L = "l",
+  Fullscreen = "fullscreen"
+}
+
+export interface ModalBaseProps {
+  children?: React.ReactNode | string;
+  /** Controls whether the modal animates in and out. ⚠️Do not use this directly⚠️ */
+  isAnimated?: boolean;
+  /** Whether the modal is open */
+  isOpen: boolean;
+  /** Which element gets focus when the modal opens */
+  initialFocus?: FocusTarget;
+  /** Function that gets called when the modal is closed */
+  onClose: (event?: React.SyntheticEvent<HTMLElement>) => void;
+  /** Which size modal to render. ⚠️Do not use this directly⚠️ */
+  size?: ModalSizes;
+}
+
+const animationDuration = 250;
+
+class ModalBase extends React.PureComponent<ModalBaseProps, {}> {
+  public static defaultProps: Partial<ModalBaseProps> = {
+    size: ModalSizes.M,
+    isAnimated: true
+  };
+
+  constructor(props) {
+    super(props);
+
+    // this.handleClose = this.handleClose.bind(this);
+    this.onKeyDown = this.onKeyDown.bind(this);
+  }
+
+  // public handleClose(e) {
+  //   console.log("handle close");
+  //   this.props.onClose(e);
+  // }
+
+  public onKeyDown(e) {
+    if (e.key === "Escape") {
+      this.props.onClose(e);
+    }
+  }
+
+  public render() {
+    const { children, isAnimated, size, initialFocus, isOpen } = this.props;
+    // TODO: clean this up, it looks weird
+    const modalSize = size ? size : ModalSizes.M;
+
+    return (
+      <Transition
```

Comment:
Should dropdownable implement this wrapper?

Reply:
I don't think so - this was just to support the modal animation